### PR TITLE
feat: fetch change request overview in project overview

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Change request info on the project overview connects to real API.

Screenshot 2024-03-22 at 16 15 11

Details:
* removed dependency on environments and using env names instead
* regenerated orval types
* exposed hook for fetching change request counts

Next PR:
* switch project insights to the new hook

### CODE REVIEW

File: frontend/src/component/project/Project/ProjectOverview.tsx [hunk 1]
```diff
@@ -39,6 +39,17 @@ const ProjectOverview: FC<{
     storageKey?: string;
 }> = ({ storageKey = 'project-overview-v2' }) => {
     const projectOverviewRefactor = useUiFlag('projectOverviewRefactor');
+
+    if (projectOverviewRefactor) {
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectOverview.tsx:43]: decided to split old and new overview with copy pasted code. eventually we'll remove the old one. My main motivation is to get rid of the project overview call in the new code

File: frontend/src/component/project/Project/ProjectOverview.tsx [hunk 2]
```diff
@@ -93,4 +96,37 @@ const ProjectOverview: FC<{
...
 };
 
+const NewProjectOverview: FC<{
+    storageKey?: string;
+}> = ({ storageKey = 'project-overview-v2' }) => {
+    const projectId = useRequiredPathParam('projectId');
+    const projectName = useProjectOverviewNameOrId(projectId);
+
+    // TODO: remove this dependency. We only need environment names
+    const { project } = useProjectOverview(projectId, {
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectOverview.tsx:106]: from this API we only care about project.environments.name so we can e.g. expose it in project insights

File: frontend/src/component/project/Project/ProjectOverviewChangeRequests.tsx [hunk 3]
```diff
@@ -42,16 +44,32 @@ const ChangeRequestCount = styled(Typography)(({ theme }) => ({
 export const ProjectOverviewChangeRequests: FC<{ project: string }> = ({
     project,
 }) => {
+    const { isChangeRequestConfiguredInAnyEnv } =
+        useChangeRequestsEnabled(project);
+    const { data: projectInsights } = useProjectInsights(project);
```
Comment (kwasniew) [inline frontend/src/component/project/Project/ProjectOverviewChangeRequests.tsx:49]: here I read all project insights only to grab change requests info. I'm also considering a dedicated resource (can be in enterprise change requests). One advantage of warming up the insights resource is that when we go to insights we already have it. The drawback is that we overload project overview with the full insights query

File: frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureToggles.tsx [hunk 4]
```diff
@@ -53,7 +51,7 @@ import { FeatureTagCell } from 'component/common/Table/cells/FeatureTagCell/Feat
 import { useSelectedData } from './hooks/useSelectedData';
 
 interface IPaginatedProjectFeatureTogglesProps {
-    environments: IProject['environments'];
+    environments: string[];
```
Comment (kwasniew) [inline frontend/src/component/project/Project/PaginatedProjectFeatureToggles/ProjectFeatureToggles.tsx:54]: relaxed coupling since we only need env names

**Review by sjaanus — APPROVED** [review]

**Pull request merged**